kubernetes/core/src/main/kotlin/com/google/kubernetes/tools/kubectl/KubectlExecutorService.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.kubernetes.tools.kubectl
+
+import com.intellij.execution.ExecutionException
+import com.google.cloud.tools.intellij.analytics.UsageTrackerService
+import com.google.common.annotations.VisibleForTesting
+import com.intellij.openapi.components.ServiceManager
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.util.concurrent.TimeUnit
+
+const val KUBECTL_FAIL = "kubectl.fail"
+const val METADATA_ERROR_MESSAGE_KEY = "error.message"
+
+class KubectlExecutorService {
+    companion object {
+        val instance
+            get() = ServiceManager.getService(KubectlExecutorService::class.java)!!
+    }
+
+    var kubectlExecutablePath: String = "kubectl"
+
+    /**
+     * run kubectl version to make sure that kubectl is available, if there's no response after
+     *  a couple of seconds then it is not available.
+     */
+    @VisibleForTesting
```

Comment:
I think we still need this method as a public API - imagine we will check the `kubectl` presence before some action or for notification - so probably it is not for tesing only?

Reply:
This check is called in getKubectlOutputBlocking, so before every command called from the API it will be ran.

Reply:
I think it might be a bit redundant to run this check on each kubectl command. If the command itself fails then we know that kubectl is not available

Reply:
@ivanporty I seem to remember you suggesting that we defer this check for a different PR?

Reply:
if a user creates a cluster from the ui from a file
kubectl create -f somefile.yaml
then it could fail either because the file is invalid or because kubectl is not available. 
This let's us know where the failure is happening.

Reply:
As discussed offline - we can keep in the `available` check.